LaTeX/LaTeX.sublime-syntax

```diff
@@ -673,6 +673,16 @@ contexts:
       push:
         - meta_scope: meta.function.citation.latex
         - include: general-optional-arguments
+        - match: '\{'
```

Comment:
This match is just a copy of the following `\{` match. If you just want to include `general-optional-arguments` before matching braces, you should just put that before the empty match pop pattern.

Reply:
This changes was made to properly recognize the following command `\parencites[Proposition~1]{Firstref}[p.~2]{SecondRef}`.

Reply:
Okay, I just noticed that the difference here is that the new version uses `push` instead of `set`. But now the match at line 686 will never match, so that should be removed.

Reply:
I might confuse something here, but the first block (with push) matches `Firstref` while the second block (with set) matches `SecondRef`.

Reply:
No, this is not how syntax definitions work. You define a set of patterns that get matched at a certain point in the file. This is a context. The patterns are checked in order. If a pattern matches, its directions are executed. If not pattern matches, the position is incremented and the process is repeated. Thus, having two identical match patterns in the same context will always lead to the first pattern winning.